src/java/htsjdk/samtools/DownsamplingIterator.java

```diff
@@ -123,6 +152,45 @@ private boolean advance() {
         return this.nextRecord != null;
     }
 
+    /**
+     * Buffers reads until either the end of the file is reached or enough reads have been buffered that
+     * downsampling can be performed and the target accuracy achieved.  Once reads have been buffered,
+     * template names are randomly sampled out for discarding until the desired number of reads have
+     * been discarded.
+     *
+     * @return True if one or more reads have been buffered, false otherwise
+     */
+    private boolean bufferNextChunkOfRecords() {
+        final int templatesToRead = (int) Math.ceil(1 / this.targetAccuracy);
+        final Set<String> names = new HashSet<String>();
+        final List<SAMRecord> recs = new ArrayList<SAMRecord>(templatesToRead);
+
+        while (this.underlyingIterator.hasNext() && names.size() < templatesToRead) {
+            final SAMRecord rec = this.underlyingIterator.next();
+            recs.add(rec);
+
+            if (!this.allowSecondaryAlignments && rec.getNotPrimaryAlignmentFlag()) continue;
```

Comment:
getNotPrimaryAlignmentFlag=true,  could also mean supplementary, right? 
what if 
```java    
    this.allowSecondaryAlignments==false;
    this.allowSupplementalAlignments==true;
```
and 
```java
    rec.getNotPrimaryAlignmentFlag()==true && rec.getSupplementaryAlignmentFlag()==true
```

I think that the record will not be included, erroneously.